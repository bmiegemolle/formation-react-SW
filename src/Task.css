.Task-todo {
}

.Task-done {
  text-decoration: line-through;
  color: #cccccc;
}
