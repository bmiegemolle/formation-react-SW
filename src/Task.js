import React from 'react';
import './Task.css';

function Task(props) {
  return (
    <div className={props.done ? "Task-done" : "Task-todo"} onClick={props.handleToggle}>
      {props.name}
    </div>
  );
}

export default Task;
