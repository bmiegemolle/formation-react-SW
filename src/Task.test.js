import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import renderer from 'react-test-renderer';
import Task from "./Task";

describe("<Task />", () => {

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Task />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  
  it('matches snapshot', () => {
    const handleToggle = jest.fn()
    const wrapper = renderer.create(<Task handleToggle={handleToggle} name='Task name' done/>).toJSON();
    expect(wrapper).toMatchSnapshot()
  })

});
