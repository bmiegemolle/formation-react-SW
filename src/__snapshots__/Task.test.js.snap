// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task /> matches snapshot 1`] = `
<div
  className="Task-done"
  onClick={[MockFunction]}
>
  Task name
</div>
`;
